---
title: "conditioned_micorbiome_QTL_manuscript"
author: "Analysis - David J. Weston, Travis Lawrence<br/>Experimentation - Sara Jawdy "
date: "11/16/2018"
output:
  html_document:
    fig_caption: yes
    number_sections: yes
    theme: readable
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

```{r setup, include=FALSE}

library(captioner)
library(knitr)
fig_nums <- captioner(prefix = "Fig.")
fig.1_cap <- fig_nums(name = "fig_1", 
                        caption = ".........")
fig.2_cap <- fig_nums(name = "fig_2", 
                        caption = ".......")
fig.3_cap <- fig_nums(name = "fig_3", 
                        caption = "......")

```

# Data exploration for conditioned microbiome analsis

## Experimental details

Sara Jawdy input

## Data extraction methods

Travis input

Output files from data extraction:
  1. phenotype_data.csv (directly from fluorescence cabinent machine)
  2. edge.Results.csv (description)
  3. edgeErode.Results.csv (description)

Notes on genotypes 
  1. all from S. fallax pedigree
  2. S.FallaxMN is the sequenced draft genome and used in Alyssa's experiments
  3. MNSA59 is the mother of the pedigree


```{r, echo=FALSE, fig.cap = fig.1_cap}
library(tidyverse)
datPhen<-read_csv("phenotype_data.csv")
datPhen$Chamber<-as.character(datPhen$Chamber) #Changing chamber from integer to character class
datEE<-read_csv("edgeErode.Results.csv") 
datEE$Chamber<-as.character(datEE$Chamber) #Changing chamber from integer to character class
datER<-read_csv("edge.Results.csv") 
datER$Chamber<-as.character(datER$Chamber) #Changing chamber from integer to character class

par(mfrow=c(1,3))
hist(datPhen$`Area(cm2)`, xlim = c(0,3), ylim = c(0,2000))
hist(datEE$`Area(cm2)`, xlim = c(0,3), ylim = c(0,2000))
hist(datER$`Area(cm2)`, xlim = c(0,3), ylim = c(0,2000))

#dat.initial.0cham0microb<-dat1 %>% filter(timepoint =="initial" & chamber =="0" & microb == "0") 

```











