---
title: "conditioned_micorbiome_QTL_manuscript"
author: "Analysis - David J. Weston, Travis Lawrence<br/>Experimentation - Sara Jawdy "
date: "11/16/2018"
output:
  html_document:
    fig_caption: yes
    number_sections: yes
    theme: readable
    toc: yes
    toc_float: yes
  pdf_document:
    toc: yes
---

```{r setup, include=FALSE}

library(captioner)
library(knitr)
fig_nums <- captioner(prefix = "Fig.")
fig.1_cap <- fig_nums(name = "fig_1", 
                        caption = "distribution of 3 differnt image extractions methods")
fig.2_cap <- fig_nums(name = "fig_2", 
                        caption = "distributions of regression model paramters using datER data")
fig.3_cap <- fig_nums(name = "fig_3", 
                        caption = "regression model relationships")

```

# Data exploration for conditioned microbiome analsis

## Experimental details

Sara Jawdy input

Experiment was initiated on 8/9/2018.
Initial data measure: 8/10/18
Week 1 data measure: 8/17/18
Week 2 data measure: 8/24/18
Week 3 data measure: 8/31/18

Notes from Sara: 3 plates of missing data; plate 12 Week1, Plate 57 week3, and plate 76 initial. 

## Data extraction methods

Travis input

Output files from data extraction:
  1. phenotype_data.csv (directly from fluorescence cabinent machine)
  2. edge.Results.csv (description)
  3. edgeErode.Results.csv (description)

Notes on genotypes 
  1. all from S. fallax pedigree
  2. S.FallaxMN is the sequenced draft genome and used in Alyssa's experiments
  3. MNSA59 is the mother of the pedigree

```{r, echo=FALSE, fig.cap = fig.1_cap}
library(tidyverse)
datPhen<-read_csv("phenotype_data.csv")
datPhen$chamber<-as.character(datPhen$chamber) #Changing chamber from integer to character class
datEE<-read_csv("edgeErode.Results.csv") 
datEE$chamber<-as.character(datEE$chamber) #Changing chamber from integer to character class
datER<-read_csv("edge.Results.csv") 
datER$chamber<-as.character(datER$chamber) #Changing chamber from integer to character class
datlocal<-read_csv("localfilter.Results.csv") 
datlocal$chamber<-as.character(datlocal$chamber) #Changing chamber from integer to character class



par(mfrow=c(2,2))
hist(datPhen$area_cm2, xlim = c(0,3), ylim = c(0,2000))
hist(datEE$area_cm2, xlim = c(0,3), ylim = c(0,2000))
hist(datER$area_cm2, xlim = c(0,3), ylim = c(0,2000))
hist(datlocal$area_cm2, xlim = c(0,3), ylim = c(0,2000))

#dat.initial.0cham0microb<-dat1 %>% filter(timepoint =="initial" & chamber =="0" & microb == "0") 

```


A couple of ideas what to test:
1) are the slopes of the data consistent across treatment (both chamber and microbe)
2) Does starting amount influnce growth tragectories? 

```{r, echo=FALSE, fig.cap = fig.2_cap}

library(broom)
#run linear gression model by genotype, micorbiome treatment and by temperature treatment

#creating new column to ID complex treatments and selecting columns to tidy the data 
lmdatainput <- datlocal %>% mutate(chamb_micro = paste(chamber, microbiome, sep = "_"))%>% select(timepoint, genotype, area_cm2, chamb_micro)

#changing the categorical 'initial', 'week1' etc to numeric days for lm
lmdatainput<- lmdatainput %>% mutate(sampletime = replace(timepoint, timepoint == "initial", 0)) %>%  mutate(sampletime = replace(sampletime, timepoint == "week1", 7)) %>% mutate(sampletime = replace(sampletime, timepoint == "week2", 14)) %>% mutate(sampletime = replace(sampletime, timepoint == "week3", 21))
lmdatainput$sampletime<-as.numeric(lmdatainput$sampletime)

#running the lm model across all rows, genotypes
lm_results<- lmdatainput %>% group_by(genotype, chamb_micro) %>% do(fitgrowth = lm(area_cm2 ~ sampletime, data = .))

#getting tidy data output from model run
lmSlopePvalue <- tidy(lm_results, fitgrowth) %>% select(genotype, chamb_micro, term, estimate, p.value) %>% filter(term =="sampletime")

lmRsquare <- glance(lm_results, fitgrowth) %>% select(genotype, chamb_micro, r.squared)

#tidy data output
lmtidyoutput<-left_join(lmSlopePvalue, lmRsquare, by = c("genotype" = "genotype", "chamb_micro" = "chamb_micro"))

# lm model parameter distributions

par(mfrow=c(1,3))
hist(lmtidyoutput$estimate, main = "lm slope")
hist(lmtidyoutput$p.value, main = "P value")
hist(lmtidyoutput$r.squared, main = "r2")

```





```{r, echo=FALSE, fig.cap = fig.3_cap}

 # lm model parameter relationships
par(mfrow=c(1,3))
plot(lmtidyoutput$p.value, lmtidyoutput$estimate, main = "p.value to slope")
plot(lmtidyoutput$r.squared, lmtidyoutput$estimate, main = "r.squared to slope")
plot(lmtidyoutput$p.value, lmtidyoutput$r.squared, main = "p.value to r.squared")
```


Gloabl patterns that include treatment (NOT genotype at this point)
NOTE: augment gives infor about the error from each point in a regression. use this to look for patterns at sampling timepoints -- is there less predicitve power at week 3 than week 1 for example? use (.resid) for this (that is difference of actual from predicted)

```{r}
library(ggplot2)

par(mfrow=c(1,3))
ggplot(lmtidyoutput, aes(x= chamb_micro, y= estimate)) +
  geom_boxplot()
ggplot(lmtidyoutput, aes(x= chamb_micro, y= r.squared)) +
  geom_boxplot()
ggplot(lmtidyoutput, aes(x= chamb_micro, y= p.value)) +
  geom_boxplot()

```








```{r}
#boxplot

  a <- augment(lm_results, fitgrowth)
  
  #str(a) says sampletime is numeric, change to factor
  
  a$sampletime<-as.factor(a$sampletime)
  
  ggplot(a, aes(color = chamb_micro, y= .resid, x= sampletime)) +
  geom_boxplot(position=position_dodge(.8)) +
  geom_jitter(position=position_dodge(.8)) +
  theme(axis.text=element_text(size=12),
        axis.text.x = element_text(angle = 90, vjust = 0.5),
        axis.title=element_text(size=16,face="bold"),
        legend.text=element_text(size=16))

```




# Get lag values for growth accumulation (growth difference from previous date. initial is set to 0)


```{r}
datlocal<- datlocal %>% mutate(cham_micro = paste(chamber, microbiome, sep = "_"))

datlocaldiff<-datlocal %>%
    group_by(genotype, cham_micro) %>%
    mutate(growth.diff = area_cm2 - lag(area_cm2, default = first(area_cm2)))

  ggplot(datlocaldiff, aes(color = cham_micro, y= growth.diff, x= timepoint)) +
  geom_boxplot(position=position_dodge(.8)) +
  geom_jitter(position=position_dodge(.8)) +
  theme(axis.text=element_text(size=12),
        axis.text.x = element_text(angle = 90, vjust = 0.5),
        axis.title=element_text(size=16,face="bold"),
        legend.text=element_text(size=16))


```


```{r}

test<-datlocaldiff %>% select(timepoint, genotype, area_cm2, cham_micro) %>% filter(timepoint == "initial" | timepoint =="week1")

datlocaldiff.week1<-test %>%
    group_by(genotype, cham_micro) %>%
    mutate(week1.diff = area_cm2 - lag(area_cm2, default = first(area_cm2))) %>% filter(timepoint =="week1")

  ggplot(datlocaldiff.week1, aes(color = cham_micro, y= week1.diff, x= cham_micro)) +
  geom_boxplot(position=position_dodge(.8)) +
  geom_jitter(position=position_dodge(.8)) +
  theme(axis.text=element_text(size=12),
        axis.text.x = element_text(angle = 90, vjust = 0.5),
        axis.title=element_text(size=16,face="bold"),
        legend.text=element_text(size=16))


  
  
  
```


